spyder/widgets/fileswitcher.py

```diff
@@ -437,7 +442,12 @@ def get_editor(self, index=None, path=None):
         elif path:
             return self.tabs.widget(index)
         else:
-            return self.parent().get_current_editor()
+            # this try is needed in order to get
+            # the current client in the notebookplugin
+            try:
+                return self.parent().get_current_editor()
+            except AttributeError:
+                return self.parent().get_current_client()
```

Comment:
Sorry, there's no need to change anything in `NotebookPlugin`. Just the first change, i.e.

```python
self.tabs.currentWidget()
```

is enough ;-)